### DISCUSSION TITLE
XQuartz

### DISCUSSION THREAD

**Discussion by GriffithLea** [asmvik/yabai#917] [Q&A] [answered]
I'm new to yabai, and I'm mostly very happy with it. I've long sought something that will mimic or replicate the focus-follows-mouse behavior that is available in the Linux/Unix world with X-Windows (X11), and this gets me almost all the way there.

My .yabairc is very simple:
yabai -m config focus_follows_mouse autofocus

With this, I get the focus behavior I want (although on "real" X11, i.e. on a Linux/Unix workstation, I also disable click-to-raise, but maybe similar behavior is not possible with MacOS+yabai). I also, I guess by default, get the behavior where I can resize windows by dragging on any corner or border, which is a boon. That's all I ask of yabai, for now at least.

However, it's not working for XQuartz. When I move the mouse away from an XQuartz window to e.g. a Finder window, the focus switches as expected, but if I then move back to an XQuartz window, the focus does not change, i.e. it remains with the previous window. Also, the XQuartz windows can only be resized with the lower right corner.

I've seen indications here and there that this is a known issue, that yabai doesn't "see" XQuartz. I was wondering if there is any workaround that would help, e.g. some .yabairc magic, even if it's not a complete solution. I just need it to work for the two features I mentioned above. Even if just the one related to focus could be made to work, that would be greatly helpful.

Finally, what is it about XQuartz that prevents it from playing nicely with yabai (or is it the other way around)? 😁

macOS 10.14.6 Mojave
yabai v3.3.10
XQuartz 2.8.1

FWIW I upgraded XQuartz from 2.7.something to 2.8.1 after I installed yabai. Probably doesn't matter, but I thought I'd better mention it.

Thanks!

**Comment by asmvik** [comment, marked as answer]
I have not dug deeply into XQuartz on macOS, but the main issue that made me stop/not attempt this integration was mainly that there is no way to listen for changes, e.g when a window is created/destroyed or otherwise manipulated. As such I figured it would be pointless to continue further, as a seamless integration was impossible, and I figured the increased complexity was not worth the unfinished/lacking support.

**Reply by GriffithLea** [reply]
Thank you. In the end, I ditched XQuartz and am running all my X11 apps inside a Linux VM that displays in the VirtualBox graphics console. I'd be happiest with Linux as my native OS, but this is a work computer, so this is what I have to do.
